### ISSUE TITLE
Suggestion: Option to raise error when record has non-enumerized value?

### ISSUE THREAD

**Issue by jzisser9** [brainspec/enumerize#453]
TLDR: A configuration option to cause Enumerize to raise an error when a record with an invalid `enumerized` attribute would be very helpful!

---

Hello! My team and I were recently investigating an issue that turned out to be due to the way Enumerize behaves with database records with an invalid value. For example, we have a database table with a `string` column, `example`, with enumerized values being `['foo', 'bar', 'baz']`. We had some records where `example` was set to `baw`. We then made an ActiveRecord query:

```ruby
Thing.where(example: 'baw').count
#=> 70

Thing.where(example: 'baw').last
<Thing:#abc123
...
example: nil
>
```

This was caused by a data migration we ran on this table before adding the `enumerize` keyword to the model. The data migration populated records in the database with a value we didn't realize at the time was invalid. So once we added Enumerize to the model, we introduced a bug that causes the behavior to break silently.

It would be really cool if there was a configuration option that causes Enumerize to raise an error when fetching a record which has an invalid value for an `enumerized` attribute. It probably makes the most sense to default it to `false` to preserve current behavior, but I'd leave that up to the implementer.

Thanks for your consideration!

**Comment by nashby** [comment]
Hey @jzisser9! Thanks for the suggestion, but it feels like an edge case. You would encounter pretty much the same issue with a regular model using, for example, an `inclusion` validation. In both cases, the record wouldn't be valid after the migration. 

I think the only difference is that with Enumerize, the `example` attribute would become `nil`, whereas with a regular `inclusion` validation, it would be `baw`. However, `record.valid?` would still return `false` either way.

Personally, I don't think this is something we need to support. During a migration, you can always run `record.valid?` or something similar to catch issues like this.
